examples/gno.land/r/demo/gnofundme/gnofundme.gno

```diff
@@ -0,0 +1,253 @@
+package gnofundme
+
+import (
+	"bytes"
+	"std"
+	"strconv"
+	"strings"
+	"time"
+
+	"gno.land/p/demo/avl"
+	pkg "gno.land/p/demo/gnofundme"
+	"gno.land/p/demo/ufmt"
+)
+
+var (
+	campaigns      *avl.Tree
+	users          *avl.Tree
+	campaignPrefix = "/r/demo/gnofundme:"
+)
+
+func init() {
+	campaigns = avl.NewTree()
+	users = avl.NewTree()
+}
+
+func NewUser(name string, email string) string {
+	// get the user address
+	address := std.GetOrigCaller()
+	// create the user
+	user := pkg.NewUser(name, email, address)
+
+	// the avl.tree key will be address to simplify the fetching
+	users.Set(address.String(), user)
+
+	return user.GetName() + " added successfully"
+}
+
+func NewCampaign(title string, description string, goal uint, beginT int64, deadlineT int64, beneficiary std.Address) (int, string) {
+	// get the caller as the owner address
+	caller := std.GetOrigCaller()
+	user := pkg.NewUser("user1", "email1", caller)
+
+	// the avl.tree key will be address to simplify the fetching
+	users.Set(caller.String(), user)
+
+	user = pkg.NewUser("user2", "email2", beneficiary)
+
+	// the avl.tree key will be address to simplify the fetching
+	users.Set(beneficiary.String(), user)
+	// verify that the user exists
+	o, ok := users.Get(caller.String())
+	if !ok {
+		panic("owner not found")
+	}
+
+	// verify that the benefeciary exists
+	b, ok := users.Get(beneficiary.String())
+	if !ok {
+		panic("beneficiary not found")
+	}
+
+	// cast the owner and beneficiary to *pkg.User
+	owner := o.(*pkg.User)
+	beneficiaryU := b.(*pkg.User)
+
+	// create the time objects
+	begin := time.Unix(beginT, 0)
+	deadline := time.Unix(deadlineT, 0)
+
+	// create the campaign
+	campaign := pkg.NewCampaign(title, description, goal, begin, deadline, owner, beneficiaryU)
+
+	// the key will be the size of the tree
+	key := campaigns.Size()
+	campaigns.Set(strconv.Itoa(key), campaign)
+
+	return key, "campaign created successfully"
+}
+
+func AddContributor(campaignId string, amount uint, display bool) string {
+	// get the campaign if it exists
+	campaign, ok := campaigns.Get(campaignId)
+	if !ok {
+		return "campaign not found"
+	}
+
+	// get the user address
+	userAddress := std.GetOrigCaller()
+
+	// get the user if he exists
+	user, ok := users.Get(userAddress.String())
+	if !ok {
+		panic("user is not registered")
```

Comment:
Why `panic` here but not in line 84? I think it would be better if you add `panic` in all the possible errors that could happen.

Reply:
thanks, didn't pay attention. (modified)